### ISSUE TITLE
[Feature Request] Gyms scan only

### ISSUE THREAD

**Issue by cyan** [RocketMap/RocketMap#1154]
heya,
i'm looking for a way to get a gym scan with fresh info, thanks to some people in the discord i have a scan with -np & -nk wich make it a lot faster but i still have a lot of useless area scanned (i live in a town the kinda the size and form of central park so a hexa would get a lot of point outside).

i've been looking in -ss and ---spawnpoints-only but could find a way to do it :/.
it would be nice to have a --gyms-only ^^

thanks again for your work and good luck :)

**Comment by hundasupa** [comment]
gym range is 900m so you can just put a worker with -st 1 in a decent spot with a long enough -sd for whatever you want the refresh rate to be. 

or do you want to cover the entire town for just gyms with a single worker?

**Comment by filippounitn** [comment]
there's already gym only scan i guess.. tried some week ago just put -nk -np and the scan will be very very wide
Try again with those 2 parameters and you should ahve what you are looking for if they didn't change it

**Comment by cyan** [comment]
@hundasupa yea i would like to scan the whole area since it thin but long, so i would need something like 10-15 spot to be scanned to get all info i think :/

@filippounitn as i said i tried that already, it work fine, but lot of useless area are scanned like half the spots

**Comment by jchristi** [comment]
@cyan, you could probably dump a list of gym locations and reformat it into -ss formatted json and run a worker with -ss set to that file. Kinda hackish but it should only scan the gym locations.

**Comment by cyan** [comment]
@jchristi i tought about that but when i got my hand on the -ss json it had a "timer" value for each spot and couldn't remove it.
i wouldn't mind having to create manualy a list of spawn point to scan (would even be better since i could scan an area and not each gyms).
i'm not that much into coding, i know how to change some little stuff but i don't know the json formated or what's really needed for a -ss json :/

**Comment by jchristi** [comment]
@cyan this would be a cool feature though. I'm worried code base is getting bloated with all the features being added (search.py is getting to be a mess). I may look into working on this later but I've got a lot of other things I'd like to do first.

**Comment by cyan** [comment]
@jchristi , if you could do something like simple scan but from coords and i add -np -nk, it would scan listed in a file for everything but those 2 ;) (maybe the code would be easier that way for a first time, and could be used in other way like scanning for specific spawn spot ;) ) thanks

**Comment by Timbo925** [comment]
Just make your own fake spawnpoint.json file to be used for scanning, just set the lat/lng coords and time you want to scan. All gyms will then be updated in an area of 900m around the selected coordinates. The time should be a value between 0 and 3600. This will influence which 'second of each hour' the scan will happen. 

Json file should be formatted like:

```json
[{"lat":50.875769,"lng":4.393943,"spawnpoint_id":"47c3f6c02b1","time":1},
 {"lat":50.862324,"lng":4.415440,"spawnpoint_id":"47c3f6c02b2","time":1000},
 {"lat":50.860718,"lng":4.384638,"spawnpoint_id":"47c3f6c02b3","time":2000}]
```

Using this method you are very light on API calls as you only scan once every hour for gym updates.

**Comment by cyan** [comment]
heya again (couldn't come back for some times ^^)
@Timbo925 i tought about it but i'd like a "real time data", something updated more often :/, since it's a small country, (like 100gyms in the town (main town in the country) +like 15-20 in the countryside...) the fight over those gyms is quite fast here ^^
wouldn't it be flagged as a scanner if i set smaller time value and the worker would teleport for hundreds of km some times?
thx

**Comment by alp-bitcoin** [comment]
I'm actively working on this.  It's nearly complete.

**Comment by jchristi** [comment]
@alp-bitcoin can you ping me on Discord? Would like to chat about your work.

**Comment by arnaudverniers** [comment]
Hi guys,

I would be interested in this feature too. I have around my house in little village almost 15 gyms and something like 15 more in the biggest city near me.

So i wanted to know how can i just scan those 30 gyms every 2-3 minutes, how can it be possible?

I tried so many things that doesn't work...

`python runserver.py -a ptc -u xxx -p xxx -ss xxx.json -l "xxx, xxx" -k "xxx" -np -nk`

Thank you for helping us.

**Comment by cyan** [comment]
heya
@alp-bitcoin  did you by chance got some result on your quest? :)

**Comment by jchristi** [comment]
I was not able to get in touch with @alp-bitcoin, so I have started working on this feature myself. I've modified @TBTerra 's spawnScan app to support 900m scans for gyms and have modified the embedded clustering tool to find clusters of gyms within 900m. All I have left is to implement a `-gs` flag that accepts a json file that searches only for gyms in a 900m radius. I will submit a PR when that is ready.
